### ISSUE TITLE
Add class to some options

### ISSUE THREAD

**Issue by canda** [leocaseiro/angular-chosen#137]
How may I add a class to some options?
I find it difficult without using ngRepeat.
I want to apply a background color to some options on the list

**Comment by leocaseiro** [comment]
Hi @canda sorry for delay, I've just took this project (Read more #162)

I found this solution at [stackoverflow](http://stackoverflow.com/a/24631572/3415716) that might works for you.

First, create the directive `option-class-expr`:

#### .js
```javascript
app.directive('optionClassExpr', function ($compile, $parse) {
    var NG_OPTIONS_REGEXP = /^\s*([\s\S]+?)(?:\s+as\s+([\s\S]+?))?(?:\s+group\s+by\s+([\s\S]+?))?\s+for\s+(?:([\$\w][\$\w]*)|(?:\(\s*([\$\w][\$\w]*)\s*,\s*([\$\w][\$\w]*)\s*\)))\s+in\s+([\s\S]+?)(?:\s+track\s+by\s+([\s\S]+?))?$/;

    return {
        restrict: 'A',
        link: function optionClassExprPostLink(scope, elem, attrs) {
            var optionsExp = attrs.ngOptions;
            if (!optionsExp) return;

            var match = optionsExp.match(NG_OPTIONS_REGEXP);
            if (!match) return;

            var values = match[7];
            var classExpr = $parse(attrs.optionClassExpr);

            scope.$watchCollection(function () {
                return elem.children();
            }, function (newValue) {
                angular.forEach(newValue, function (child) {
                    var child = angular.element(child);
                    var val   = child.val();
                    if (val) {
                        child.attr('ng-class', values + '[' + val + '].' +
                                               attrs.optionClassExpr);
                        $compile(child)(scope);
                    }
                });
            });
        }
    };
});
```

Than, you can use like that
#### .html
```html
<select ng-model="someObj" ng-options="obj as obj.color for obj in objects"
        option-class-expr="color">
</select>
```

**Comment by stetsenko** [comment]
@leocaseiro thanks for taking care of chosen!

It would be nice to assign specific class for each option. For example, in order to have a different font awesome icon for each option. Basically similar to what angular-strap#select allows:

<img width="372" alt="screen shot 2016-05-27 at 5 39 38 pm" src="https://cloud.githubusercontent.com/assets/732110/15624003/03ee93e6-2432-11e6-8876-e1158f0b7cfa.png">

Is that possible?

**Comment by leocaseiro** [comment]
Hi @stetsenko,

Thank you for your supporting.

I've just realised my solution didn't work.It creates classes for options, but chosen is not receiving it.

I did some research to use `ng-class` or similar for each `option` with `ng-options` which is an angular issue. It's not easy. In angular-strap, they use HTML for the select label.

However, chosen don't accept icons well.
I honestly don't know what can do to help you out =/

Unfortunately, I'll try to look deeply in a later future.
